src/main/java/com/passbook/sparkeighteen/peristence/POJO/SignUpRequest.java

```diff
@@ -1,24 +1,37 @@
 package com.passbook.sparkeighteen.peristence.POJO;
 
+import com.passbook.sparkeighteen.peristence.POJO.Gender;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
 import javax.validation.constraints.NotBlank;
-import javax.validation.constraints.NotEmpty;
+import java.time.LocalDate;
 
 @Data
 @Builder
 @AllArgsConstructor
 @NoArgsConstructor
 public class SignUpRequest {
 
-    @NotEmpty(message = "email is mandatory")
-    private String email;
+    @NotBlank(message = "First name cannot be left Blank")
+    private String firstname;
+
+    private String lastname;
+
+    @NotBlank(message = "Gender cannot be left Blank")
+    @Enumerated(EnumType.STRING)
+    private Gender gender;
 
-    @NotBlank
-    @NotEmpty(message = "password is mandatory")
+    @NotBlank(message = "Date of Birth of a user cannot be blank")
+    private LocalDate dob;
+
+    @NotBlank(message = "Password cannot be blank")
     private String password;
 
+    @NotBlank(message = "Email cannot be blank")
+    private String email;
```

Comment:
Just one question are almost all fields mandatory during signup it-self ?
Think about It just.

Reply:
All fields are mandatory as they are necessary fields to build a profile.